### DISCUSSION TITLE
Is there away to choose only the first of multiple matches?

### DISCUSSION THREAD

**Discussion by haphamdev** [junegunn/fzf#3039] [Q&A]
Example:
I have a file call `text.txt` as below
```
> cat text.txt
one
two
three
```

Then `cat text.txt | fzf --filter o` will produce two matches
```
> cat text.txt | fzf --filter o
one
two
```

Is there any way to make `fzf --filter` to return only the first match?

Something like:
```
> cat text.txt | fzf --filter --first-match-only o
one
```

**Comment by haphamdev** [comment]
Of course I can try `cat text.txt | fzf --filter o | head -1`. 
However, **in case of no match**, the status is still 0 as if there were no error.

Demo:
Without `head -1`, `$?` will be `1` which means error
```
bash-5.2$ cat text.txt | fzf --filter abc
bash-5.2$ echo $?
1
```

With `head -1`, `$?` will be `0` which means OK
```
> cat text.txt | fzf --filter abc | head -1
> echo $?
0
```

**Reply by LangLangBart** [reply]
> However, **in case of no match**, the status is still 0 as if there were no error.

[GNU Bash Reference Manual](https://www.gnu.org/software/bash/manual/bash.html)
> The exit status of a pipeline is the exit status of the last command in the pipeline, unless the pipefail option is enabled (see [The Set Builtin](https://www.gnu.org/software/bash/manual/bash.html#The-Set-Builtin)). If pipefail is enabled, the pipeline’s return status is the value of the last (rightmost) command to exit with a non-zero status, or zero if all commands exit successfully. If the reserved word ‘!’ precedes the pipeline, the exit status is the logical negation of the exit status as described above. The shell waits for all commands in the pipeline to terminate before returning a value.

```zsh
set -o pipefail
```
![them](https://user-images.githubusercontent.com/92653266/200546700-ec22302a-b8da-4579-abfd-67def5257ac2.gif)

---
